src/button-group/internal.tsx

```diff
@@ -0,0 +1,297 @@
+        return;
+      }
+
+      if (tooltipItemId !== itemId) {
+        if (show) {
+          setTooltipItemId(itemId);
+        }
+      } else {
+        if (!show) {
+          setTooltipItemId(null);
+        }
+      }
+
+      setIsFeedbackTooltip(false);
+    };
+
+    const tooltipItem = tooltipItemId ? allItems[tooltipItemId] : undefined;
+
+    return (
+      <div
+        {...baseProps}
+        className={clsx(styles.root, testUtilStyles['button-group'], baseProps.className)}
+        ref={containerRef}
+        role="toolbar"
+        aria-label={ariaLabel}
+        onFocus={onFocus}
+        onBlur={onBlur}
+        onKeyDown={onKeyDown}
+      >
+        <SingleTabStopNavigationProvider
+          ref={navigationAPI}
+          navigationActive={true}
+          getNextFocusTarget={getNextFocusTarget}
+          onUnregisterFocusable={onUnregisterFocusable}
+        >
+          {items.map((itemOrGroup, index) => {
+            const itemContent = (item: ButtonGroupProps.Item) => (
+              <div
+                key={item.id}
+                className={styles['item-wrapper']}
+                ref={element => (itemWrappersRef.current[item.id] = element)}
+                onPointerEnter={() => onShowTooltip(item.id, true)}
+                onPointerLeave={() => onShowTooltip(item.id, false)}
+                onFocus={() => onShowTooltip(item.id, true)}
+                onBlur={() => onShowTooltip(item.id, false)}
+              >
+                <ItemElement
+                  item={item}
+                  dropdownExpandToViewport={dropdownExpandToViewport}
+                  onItemClick={event => onClickHandler(item.id, event)}
+                  ref={element => (itemsRef.current[item.id] = element)}
+                />
+              </div>
+            );
+
+            const isGroupBefore = items[index - 1]?.type === 'group';
+            const currentItem = items[index];
+            const isGroupNow = currentItem?.type === 'group';
+            const shouldAddDivider = isGroupBefore || (!isGroupBefore && isGroupNow && index !== 0);
+
+            if (isGroupNow && currentItem.items.length === 0) {
+              warnOnce('ButtonGroup', 'Empty group detected. Empty groups are not allowed.');
+            }
+
+            return (
+              <React.Fragment key={itemOrGroup.type === 'group' ? index : itemOrGroup.id}>
+                {shouldAddDivider && <div className={styles.divider} />}
+                {itemOrGroup.type === 'group' ? (
+                  <div key={index} role="group" aria-label={itemOrGroup.text} className={styles.group}>
+                    {itemOrGroup.items.map(item => (
+                      <React.Fragment key={item.id}>{itemContent(item)}</React.Fragment>
+                    ))}
+                  </div>
+                ) : (
+                  <React.Fragment key={itemOrGroup.id}>{itemContent(itemOrGroup)}</React.Fragment>
```

Comment:
This fragment is probably not needed here

Reply:
Removed